app/infrastructure/persistence/repositories/user_repo.py

```
@@ -115,8 +127,20 @@ async def create_user(
+            return _user_to_result(created)
+        except IntegrityError:
+            raise UserAlreadyExistsException()
+
+    async def update(
+        self, obj: User, *, skip_existence_check: bool = False
+    ) -> User:
+        """Update user; raise DuplicateEmailException on unique constraint (e.g. duplicate email)."""
+        try:
+            return await super().update(obj, skip_existence_check=skip_existence_check)
+        except IntegrityError:
+            raise DuplicateEmailException()
```

Ruff (0.15.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)